### ISSUE TITLE
Map never stops tracking user location

### ISSUE THREAD

**Issue by Spacarar** [Yermo/nativescript-mapbox#312]
#### Description of workflow
- The user will drill down a series of menus to select an area to navigate
- Once a user has selected the region they wish to view, they are taken to a page with a `<Mapbox>` map that has `trackUserLocation="true"`

#### Current result
If a user navigates from the map page, back to the list, then back to the map page, the message `delivering location to client's delegate` is repeated for each visit. thus visiting three different regions will repeat this message three times each time the location is delivered.

This happens even when setting `trackUserLocation = false` or using `destroy()`

#### Expected result
Once a user leaves the page that requires their user location, the map stops tracking the user (even if this has to be done manually)

**Comment by lschmidlin** [comment]
Agreed. I see this on IOS. In my case, I am doing destroy() on the map when leaving a page with it, but still see the debugging messages of "delivering location to client's delegate" in the debugger.

**Comment by zencm** [comment]
For me this also is the case when initiating and canceling the tracking behavior programmatically.
Still seeing the messages after `destroy()` or `trackUser({mode:'NONE'})`

**Comment by Spacarar** [comment]
@zencm well `trackUser({mode: 'NONE'})` actually starts tracking with no heading or course. Still an issue though.
